test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -96,10 +95,10 @@ public void MatrixFactorizationSimpleTrainAndPredict()
             // MF produce different matrices on different platforms, so check their content on Windows.
```

Comment:
Very cool finding and fix Mustafa!

Could you please add comments to explain the reason why we need different metrics for different os, also attach your fix to libmf for better reference in future?

Reply:
Thank you Frank! Of course, I'll add explanatory comments to the codebase. I've already attached the libmf PR fixes into this PR's explanation, and will add related issues to it too.